tests/test_github.ml

```diff
@@ -818,37 +838,99 @@ let test_contexts = [|
   ["ci"; "circleci"];
 |]
 
+let test_pr_commits =  [| "123"; "456"; "789"; "0ab" |]
+let test_ref_commits = [| "123"; "456"; "abc"; "def" |]
+
+let test_names = [|
+  ["heads"; "master"];
+  ["tags" ; "foo"; "bar"];
+  ["heads"; "gh-pages"];
+|]
+
+let test_descriptions = [|
+  Some "Testing...";
+  None;
+|]
+
+let test_state = [|
+  `Pending;
+  `Success;
+  `Failure;
+  `Error;
+|]
+
 let random_choice ~random options =
   options.(Random.State.int random (Array.length options))
 
-let random_state ~random ~user ~repo ~old_prs ~old_commits =
+let random_pr_commit ~random ~repo =
+  let id = random_choice ~random test_pr_commits in
+  { Commit.repo; id }
+
+let random_ref_commit ~random ~repo =
+  let id = random_choice ~random test_ref_commits in
+  { Commit.repo; id }
+
+let random_description ~random = random_choice ~random test_descriptions
+
+let random_state ~random = random_choice ~random test_state
+
+let random_refs ~random ~repo ~old_refs =
+  test_names
+  |> Array.to_list
+  |> List.map (fun name ->
+      if Random.State.bool random then (
+        match List.find (fun r -> r.Ref.name = name) old_refs with
```

Comment:
I don't think you need `old_refs` here. GitHub is fine with deleting existing refs.